django/forms/widgets.py

```diff
@@ -79,8 +77,8 @@ def absolute_path(self, path, prefix=None):
 
     def __getitem__(self, name):
         "Returns a Media object that only contains media of the given type"
-        if name in MEDIA_TYPES:
-            return Media(**{str(name): getattr(self, '_' + name)})
+        if name in [mt[0] for mt in self.MEDIA_TYPES]:
```

Comment:
The size of self.MEDIA_TYPES won't be large, but it would be good to pre-compute this.